frame/nomination-pools/src/lib.rs

```diff
@@ -661,30 +757,10 @@ impl<T: Config> BondedPool<T> {
 		sub_pools: &SubPools<T>,
 	) -> Result<bool, DispatchError> {
 		if *target_account == self.roles.depositor {
-			// This is a depositor
-			if !sub_pools.no_era.points.is_zero() {
-				// Unbonded pool has some points, so if they are the last delegator they must be
-				// here. Since the depositor is the last to unbond, this should never be possible.
-				ensure!(sub_pools.with_era.len().is_zero(), Error::<T>::NotOnlyDelegator);
-				ensure!(
-					sub_pools.no_era.points == target_delegator.points,
-					Error::<T>::NotOnlyDelegator
-				);
-			} else {
-				// No points in the `no_era` pool, so they must be in a `with_era` pool
-				// If there are no other delegators, this can be the only `with_era` pool since the
-				// depositor was the last to withdraw. This assumes with_era sub pools are destroyed
-				// whenever their points go to zero.
-				ensure!(sub_pools.with_era.len() == 1, Error::<T>::NotOnlyDelegator);
-				sub_pools
-					.with_era
-					.values()
-					.next()
-					.filter(|only_unbonding_pool| {
-						only_unbonding_pool.points == target_delegator.points
-					})
-					.ok_or(Error::<T>::NotOnlyDelegator)?;
-			}
+			ensure!(
```

Comment:
@emostov correct?

Reply:
Yes, should be enough.

We also need to check that the depositor has no active bonded, but as long we don't let the depositor partial unbond we should be good .. maybe we should add a note mentioning this